target/linux/brcm47xx/image/Makefile

```diff
@@ -536,146 +560,153 @@ endif
 
 ifeq ($(SUBTARGET),mips74k)
 define Device/asus-rt-ac53u
-  DEVICE_TITLE := Asus RT-AC53U
+  DEVICE_MODEL := RT-AC53U
   DEVICE_PACKAGES := $(USB2_PACKAGES)
   $(Device/asus)
   PRODUCTID := RT-AC53U
 endef
 TARGET_DEVICES += asus-rt-ac53u
 
 define Device/asus-rt-ac66u
-  DEVICE_TITLE := Asus RT-AC66U
+  DEVICE_MODEL := RT-AC66U
   DEVICE_PACKAGES := kmod-b43 $(USB2_PACKAGES)
   $(Device/asus)
   PRODUCTID := RT-AC66U
 endef
 #  TARGET_DEVICES += asus-rt-ac66u
 
 define Device/asus-rt-n10
-  DEVICE_TITLE := Asus RT-N10
+  DEVICE_MODEL := RT-N10
   DEVICE_PACKAGES := kmod-b43
   $(Device/asus)
   PRODUCTID := "RT-N10      "
 endef
 TARGET_DEVICES += asus-rt-n10
 
 define Device/asus-rt-n10p
-  DEVICE_TITLE := Asus RT-N10P
+  DEVICE_MODEL := RT-N10P
+  DEVICE_VARIANT := v1
   DEVICE_PACKAGES := kmod-b43
   $(Device/asus)
   PRODUCTID := RT-N10P
 endef
 TARGET_DEVICES += asus-rt-n10p
 
 define Device/asus-rt-n10p-v2
-  DEVICE_TITLE := Asus RT-N10P v2
+  DEVICE_MODEL := RT-N10P
+  DEVICE_VARIANT := v2
   $(Device/asus)
   PRODUCTID := RT-N10PV2
 endef
 TARGET_DEVICES += asus-rt-n10p-v2
 
 define Device/asus-rt-n10u
-  DEVICE_TITLE := Asus RT-N10U
+  DEVICE_MODEL := RT-N10U
   DEVICE_PACKAGES := kmod-b43 $(USB2_PACKAGES)
```

Comment:
If you want to add implicit revisions, then the wrt54gs should also get the v1. I feel a bit unsure about this approach in general.

Reply:
Actually, I asked myself the very same question when looking at this PR an hour ago.

My general opinion on that is: Yes.
Thinking about how this is leveraged in the image selector, one should always have VARIANT set for all devices having same VENDOR/MODEL combination.
(It would also help if this was considered more frequently when adding devices in the first place.)

For the question at hand: I do not know whether `DEVICE_VARIANT := A` is correct here, as I'm not familiar with these devices (so, is it "A"?) But I'd would say it's desirable to find out and add the VARIANT.

Reply:
There is no explicit A variant: https://wikidevi.com/wiki/ASUS_RT-N10U

Reply:
btw. I think the image selector should make the choice to add implicit revisions.

Reply:
What does "explicit" and "implicit" mean in your terminology?

I explicitly think it is not a good idea to have the image selector "calculate" revisions. One would have to have an ugly script then which looks at similar devices and e.g. derives "v1" if it finds a "v2". Will it work for "v 2" or "v. 2"?
If we do not set VARIANT here, I would recommend to also have something like "(no rev.)" or "(none)" in the VARIANT field of the image selector.

Under those circumstances, I could easier accept having revisions "no rev." and "B" for the RT-N10U than having "no rev." and "v2" for some e.g. TP-Link device. I still would add the "A" here, if one asked me, but I could live without the "A" here much easier than I would like the Linksys having no "v1".

Reply:
explicit: used by the vendor (on the website or printed on the casing/board)
implicit: we conclude that it is e.g. v1 or A, because there is a later model that uses v2 or B

Reply:
@adrianschmutzler but now we calculate revisions ourselves. The point of this change is to enable a machine readable output, but we should refrain from adding dummy data (e.g. v1) for that purpose to the raw/source of data that are these Makefiles.
Can you maybe get a third opinion? If some dev says that is ok, I will accept it. But I feel I have to push a back a bit when I feel that we are off the track.

Reply:
Well, I won't actively try to _get_ a third opinion, as I'm only expressing my personal opinion here without any authority. So you are free to ignore it anyway. ;-)
But I agree that at this point of discussion we won't solve it by discussing further, but can only proceed by having someone else's opinion.
(This might also be a committer just merging it, which would however only give a hint for similar discussions elsewhere).

Reply:
ok ]:->

Reply:
@lynxis can you help to resolve this or refer to somebody else?